src/TestLogger/Core/TestRunResultWorkflow.cs

```diff
@@ -22,11 +24,24 @@ public static void Result(this ITestRun testRun, TestResultEventArgs resultEvent
                 _ => Parser.Parse(fqn),
             };
 
+            var result = resultEvent.Result;
+            Func<string, string> sanitize = testRun.Serializer.InputSanitizer.Sanitize;
```

Comment:
Before I do the remaining polish, let me know what you think of this approach overall. My thought was that if we handle data cleaning in the workflows, the loggers won't need any changes, except picking which `IInputSanitizer` they need. 

I'm being a bit aggressive about sanitizing every string we can get in result info, but I figured it might be easier in the long run.

Reply:
This looks good to me. Let's add a no-op sanitizer that simply returns the input in case a logger doesn't set the sanitizer.

Reply:
Do we want a default? I think all the loggers that use this one should use the Xml sanitizer.

Reply:
We don't have a way to enforce this assumption that all `ITestResultSerializer` implementations must provide a valid `IInputSanitizer`. It may be a good fallback to add an identity function.

`Func<string, string> sanitize = testRun.Serializer.InputSanitizer?.Sanitize ?? (x) => x;` might work.

Reply:
I'm not seeing any risk here. If we forgot to setup the serializer in one of the downstream loggers, we would just end up failing the CI build.

Reply:
Sounds reasonable.